obspy/taup/seismic_phase.py

```diff
@@ -146,37 +152,68 @@ def parse_name(self, tMod):
         currLeg = self.legs[0]
         nextLeg = currLeg
         isPWave = True
+        isPWavePrev = isPWave
```

Comment:
I was mostly following the Java version, but I think you're right; it can be moved to where `currLeg` is set before the loop.